src/webgpu/api/operation/memory_sync/buffer.spec.ts

```diff
@@ -0,0 +1,257 @@
+        module: this.device.createShaderModule({
+          code: wgslFragment,
+        }),
+        entryPoint: 'frag_main',
+      },
+      primitiveTopology: 'point-list',
+      colorStates: [{ format: 'rgba8unorm' }],
+    });
+  }
+
+  beginSimpleRenderPass(encoder: GPUCommandEncoder): GPURenderPassEncoder {
+    const view = this.device
+      .createTexture({
+        size: { width: 4, height: 4, depth: 1 },
+        format: 'rgba8unorm',
+        usage: GPUTextureUsage.OUTPUT_ATTACHMENT,
+      })
+      .createView();
+    return encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: view,
+          loadValue: { r: 0.0, g: 1.0, b: 0.0, a: 1.0 },
+          storeOp: 'store',
+        },
+      ],
+    });
+  }
+
+  // Create render pass or compute pass, set pipeline and bind groups and write data into storage
+  // buffer. If the pass is render pass, it may use bundle. Note that we write 1 into the buffer
+  // if the pass is used in the first write, and we write 2 into the buffer otherwise.
+  createPass(
+    buffer: GPUBuffer,
+    bundle: boolean,
+    isRenderPass: boolean,
+    encoder: GPUCommandEncoder,
+    isFirstWrite: boolean
+  ) {
+    assert(!bundle || isRenderPass);
+    if (isRenderPass) {
+      const pipeline = this.createRenderPipeline(isFirstWrite);
+      const bindGroup = this.createBindGroup(pipeline, buffer);
+      const pass = this.beginSimpleRenderPass(encoder);
+      if (bundle) {
+        const bundleEncoder = this.device.createRenderBundleEncoder({
+          colorFormats: ['rgba8unorm'],
+        });
+        bundleEncoder.setBindGroup(0, bindGroup);
+        bundleEncoder.setPipeline(pipeline);
+        bundleEncoder.draw(1, 1, 0, 0);
+        const bundle = bundleEncoder.finish();
+        pass.executeBundles([bundle]);
+      } else {
+        pass.setPipeline(pipeline);
+        pass.setBindGroup(0, bindGroup);
+        pass.draw(1, 1, 0, 0);
+      }
+      pass.endPass();
+    } else {
+      const pipeline = this.createComputePipeline(isFirstWrite);
+      const bindGroup = this.createBindGroup(pipeline, buffer);
+      const pass = encoder.beginComputePass();
+      pass.setPipeline(pipeline);
+      pass.setBindGroup(0, bindGroup);
+      pass.dispatch(1);
+      pass.endPass();
+    }
+  }
+
+  createCommandBuffer(
+    buffer: GPUBuffer,
+    bundle: boolean,
+    isRenderPass: boolean,
+    isFirstWrite: boolean
+  ): GPUCommandBuffer {
+    const encoder = this.device.createCommandEncoder();
+    this.createPass(buffer, bundle, isRenderPass, encoder, isFirstWrite);
+    return encoder.finish();
+  }
+
+  verifyData(buffer: GPUBuffer, expectedValue: number) {
+    const bufferData = new Uint32Array(1);
+    bufferData[0] = expectedValue;
+    this.expectContents(buffer, bufferData);
+  }
+}
+
+export const g = makeTestGroup(BufferSyncTest);
+
+// Test write-after-write between passes. The first write will write 1 into a storage buffer.
```

Comment:
nit: it's `g.test().desc()` which is per-test (g.desc would be per-file which is the same as the existing `description`)

Reply:
Sounds like we don't need to change for the description part.

Reply:
What I meant is that the content of this comment can be put in `g.test().desc()` so that it appears in the harness (and Kai pointed out I mistakenly said `g.desc()`). This is a new feature of `g.test()` that isn't used in any test but that AFAIK Kai would like to move towards having used eveywhere.

Reply:
Done.